providers/oauthProvider.js

```diff
@@ -130,3 +132,35 @@ exports.makeUserSiteAdmin = (externalUserId, oauthDefaultId) => {
       body: JSON.stringify(body)
     });
 };
+
+
+exports.copyUsersFromSite = async (oldSiteId, newSiteId) => {
+  const roleResponse = await userRoleApi.fetchAll();
+  const roleMap = new Map(roleResponse.map(role => [role.name, role.id]));
+  const newSite = await siteApi.fetch(newSiteId);
+  const newClientId = newSite.config?.oauth?.default?.id;
+  const usersOfOldSite = await siteApi.fetchUsersOfSite(oldSiteId);
```

Comment:
Since users (actually: userroles) are copied to the oauth server they should be fetched from the oauth server, not from the API.

Reply:
Furthermore you should copy much less than is done now:
1. it is not necessary to copy users with the default role; that is automatically added when a user tries to login
2. anonymous users should probably not be copied